Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -582,6 +584,12 @@ int AXTextMarker::lineNumberForIndex(unsigned index) const
     auto* stopObject = object->nextUnignoredSiblingOrParent();
     auto stopAtID = stopObject ? std::make_optional(stopObject->objectID()) : std::nullopt;
 
+    // Mimic behavior of AccessibilityRenderObject::visiblePositionForIndex.
+    auto objectRange = object->textMarkerRange();
+    unsigned textControlLength = objectRange.toString().length();
+    if (object->isTextControl() && index >= textControlLength)
+        return -1;
```

Comment:
Seems like we only use the result of `AXTextMarkerRange::toString()` when `object->isTextControl()`. I think it might be more readable and more performant to do something like this?

```cpp
if (object->isTextControl() && index >= object->textMarkerRange().toString().length()) {
    // Mimic behavior of AccessibilityRenderObject::visiblePositionForIndex.
    return -1;
}
```

Reply:
Good idea!